string msg_sent
---
string msg_feedback
